Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealitySpatialMeshObserver.cs

```diff
@@ -677,13 +677,18 @@ private void SurfaceObserver_OnDataReady(SurfaceData cookedData, bool outputWrit
                 bool sendUpdatedEvent = false;
                 if (meshes.ContainsKey(cookedData.id.handle))
                 {
-                    // Reclaim the old mesh object for future use.
-                    ReclaimMeshObject(meshes[cookedData.id.handle]);
-                    meshes.Remove(cookedData.id.handle);
+                    SpatialAwarenessMeshObject toRemove = meshes[cookedData.id.handle];
 
+                    meshes[cookedData.id.handle] = meshObject;
```

Comment:
Can you explain a little more about how the fix is addressing the linked issue #8097?

It looks like the only real delta in this change is that we're inserting the meshObject into the meshes dictionary first and then reclaiming the object, vs before where we deleted it first and then re-added it to the meshes dictionary. However, the meshes dictionary is just a local data structure so it's not like adding anything there will trigger anything interesting, so I don't fully understand how this order actually changes anything

Reply:
After reviewing the code for the observer, i did not find anything that was obviously "deleting immediately and adding after a delay" this was an attempt to see if not calling the add path (just updating in place) made a difference.

Otherwise, i believe the issue in #8097 is lower than MRTK (Unity or platform behavior related?)

Reply:
I think @matthejo's comment he mentioned a specific codepath where the removes are done immediately but the adds are async - it seems like that could likely be the case. Was that particular path ruled out?